Packs/Netskope/Integrations/NetskopeEventCollector/NetskopeEventCollector.py

```diff
@@ -138,21 +138,31 @@ def get_events_v1(client: Client, last_run: dict, limit: Optional[int] = None) -
             if response.get('status') != 'success' or not (results := response.get('data', [])):  # type: ignore
                 break
 
-            et_events.extend(results)
-
+            events = []
+            for event in results:
+                if event.get('timestamp') == last_run[event_type] and event.get('_id') not in last_run_ids:
+                    events.append(event)
+                    last_run_ids.add(event['_id'])
+                else:
+                    last_run[f'{event_type}-ids'] = [event['_id']]
+                    last_run_ids = set(last_run[f'{event_type}-ids'])
```

Comment:
You're setting last_run[f'{event_type}-ids'] further down.
```suggestion
                    last_run_ids = set(event['_id'])
```